Dev refactoring

kripso opened the pull request:
Updated some thigns that were driving me crazy while i tried refactoring to FastAPI...
includes changes to the flow of the app.. now you can go easily back and forth the analysis.

analysis table is still a placeholder but the flow is now bit more adjusted to the one it could be.

fixed tabs styling and run npm format on the whole project..

renamed id's to their respective names node_id and session_id.. there is still some mixup with what should be named how.. but i can live with this for now.

updated the .proto file accordingly..

web_api now wont fails in some cases on hitting preview

kripso committed:
WIP refactoring

kripso committed:
fixed main page

kripso committed:
fixed checks

stefhajdu committed:
fixed expected value in test_load_sessions unit test

stefhajdu committed:
one more test run

stefhajdu commented:
fixed my random unit test, continue with reviewing...

stefhajdu commented on client_app/src/routes/analyses/+page.svelte:
```diff
@@ -1,23 +1,83 @@
   await globalAnalysesState.setAnalysisFromAPI();
 });
+
+$effect(() => {
```
could you use selectedIndex = [$derived.by(() => {})](https://svelte.dev/docs/svelte/$derived#$derived.by) instead of $effect, or it won't work? It should react on changes made in globalAnalysesState, which is already state...

kripso replied:
actually nothing was needed.. fixed it now with bit of styling on the buttons

stefhajdu commented on client_app/src/routes/analyses/+page.svelte:
```diff
@@ -1,23 +1,83 @@
-    {/each}
-  </Tabs>
+
+  <div class="sticky z-10 flex items-center gap-4 px-4 py-3">
```
I like this, those Tabs were terrible

kripso committed:
fixed

kripso committed:
Changes to be committed:
	modified:   client_app/.svelte-kit/ambient.d.ts
	modified:   client_app/.svelte-kit/generated/client/app.js
	modified:   client_app/.svelte-kit/generated/client/matchers.js
	modified:   client_app/.svelte-kit/generated/client/nodes/0.js
	modified:   client_app/.svelte-kit/generated/client/nodes/1.js
	modified:   client_app/.svelte-kit/generated/client/nodes/2.js
	modified:   client_app/.svelte-kit/generated/client/nodes/3.js
	modified:   client_app/.svelte-kit/generated/root.js
	modified:   client_app/.svelte-kit/generated/server/internal.js
	modified:   client_app/.svelte-kit/non-ambient.d.ts
	modified:   client_app/.svelte-kit/tsconfig.json
	modified:   client_app/.svelte-kit/types/route_meta_data.json
	modified:   client_app/.svelte-kit/types/src/routes/$types.d.ts
	modified:   client_app/.svelte-kit/types/src/routes/analyses/$types.d.ts
	modified:   client_app/package.json
	modified:   client_app/src/components/Footer.svelte
	modified:   client_app/src/components/Header.svelte
	modified:   client_app/src/components/Nodes/NodeClass.svelte.ts

kripso committed:
Changes to be committed:
	modified:   client_app/.prettierrc
	modified:   client_app/src/components/DataFrameTable.svelte
	modified:   client_app/src/components/Expression/AddColumnExpressionFrom.svelte
	modified:   client_app/src/components/Header.svelte
	modified:   client_app/src/components/Nodes/LoadNode/LoadNode.svelte
	modified:   client_app/src/routes/+layout.svelte
